src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy

```diff
@@ -805,15 +801,12 @@ public enum LegendsAwakened implements LogicCardInfo {
               if (r==PLAY_FROM_HAND && confirm("Use Time Walk?")){
                 assert my.hand : "No cards in hand"
                 powerUsed()
-                def tar = my.prizeCardSet.faceDownCards.select(hidden: false, min: 0, "Choose a Pokemon in your prizes to replace with a card in your hand.").first()
-                assert tar : "No Pokemon selected."
-                assert tar.hasTypes(POKEMON) : "Only Pokemon can be grabbed by Time Walk."
-                def newPrize = my.hand.select(hidden: true, "Card to put into Prizes").first()
-                def indexOfOldPrize = my.prizeCardSet.indexOf(tar)
-                my.hand.add(tar)
-                my.prizeCardSet.set(indexOfOldPrize, newPrize)
-                my.prizeCardSet.setVisible(newPrize, false)
-                my.hand.remove(newPrize)
+                def tar = my.prizeCardSet.faceDownCards.cardTypeFilter(POKEMON).select(hidden: false, min: 0, "Choose a Pokemon in your prizes to replace with a card in your hand.").first()
+                if (tar) {
+                  def newPrize = my.hand.select(hidden: true, "Card to put into Prizes").first()
+                  newPrize.moveTo(hidden: true, my.prizeCardSet)
+                  tar.moveTo(my.hand)
```

Comment:
> Q. Can I rearrange the order of my prize cards after using Azelf's "Time Walk" Poke-POWER?
A. Yes, you can. (Oct 16, 2008 PUI Rules Team)

Might be more true to the game to allow a user to rearrange prize cards after performing these actions. Though it might be overkill and be an option players don't often choose to use too. Might be worth discussing in the Discord.

Reply:
We'll table whether you are supposed to rearrange only face down cards, or all prize cards for now until we can find a proper ruling.